altos-rust/altos-core/src/sched/mod.rs

```diff
@@ -86,33 +88,52 @@ pub fn switch_context() {
 
 ///Get the next task from the queue
 fn schedule_next_task() -> bool {
-  let mut queues = Priority::all();
-  unsafe {
-    if CRITICAL_TASK_COUNTER >= CRITICAL_TASK_MAX {
-      queues = 1..NUM_PRIORITIES;
-      CRITICAL_TASK_COUNTER = 0;
-    }
+  let queues = Priority::all();
+  if NORMAL_TASK_COUNTER.load(Ordering::Relaxed)  >= NORMAL_TASK_MAX {
+    NORMAL_TASK_COUNTER.store(0, Ordering::Relaxed);
+    return schedule_next_task_rigged();
```

Comment:
So I don't think this is the right way to approach this problem. I realize the `all_except` method returns a different type than the `all` method, but this would be a good place to factor out the function into a generic so we don't have to repeat the scheduling code in two places. The functions signature would probably look something like: 
```rust
fn schedule_task<T>(queues: T) where T: Iterator<Item=usize> {
    // Scheduling code...
}
```

This way we can decide if we should grab all the queues or filter some, then just pass the sequence into this one function.